crates/db-address-space/src/address_space.rs

```diff
@@ -194,21 +181,19 @@ impl AddressSpaceRegion {
     /// * `size` - Length of content to map
     /// * `mem_type` - Memory mapping from, 'shmem' or 'hugetlbfs'
     /// * `mem_file_path` - Memory file path
-    /// * `numa_node_id` - NUMA node id to allocate memory from
-    /// * `mem_prealloc_enabled` - Enable prealloc of guest memory or not
-    /// * `region_type` - The type of address spacee region
-    #[allow(clippy::too_many_arguments)]
+    /// * `numa_node_id` - Optional NUMA node id to allocate memory from
+    /// * `mem_prealloc` - Whether to enable pre-allocation of guest memory
+    /// * `is_hotplug` - Whether it's a region for hotplug.
     pub fn create_memory_region(
         base: GuestAddress,
         size: GuestUsize,
         mem_type: &str,
         mem_file_path: &str,
         numa_node_id: Option<u32>,
-        mem_prealloc_enabled: bool,
+        mem_prealloc: bool,
```

Comment:
Can we use a enum to represent memory types  instead of &str? Which is more precise? Or use a trait to represent memory type, which can simplify the build process, and have more expandability?

Reply:
deserve a dedicated PR.